lib/logger/logger.js

```diff
@@ -15,7 +15,7 @@ Logger.prototype.setMethods = function (overrides) {
 
   SUPPORTED_METHODS.forEach(function (method) {
     if (typeof overrides[method] === 'function') {
-      logger[method] = overrides[method];
+      logger[method] = overrides[method].bind(overrides);
```

Comment:
Nice, but shouldn't this be: 

```suggestion
      logger[method] = overrides[method].bind(overrides[method]);
```

And by that point, maybe we can tidy up the loop a bit:

```javascript
SUPPORTED_METHODS.forEach(function (method) {
  var override = overrides[method];
  if (typeof override === 'function') {
    logger[method] = override.bind(override);
  }
});
```

Reply:
should be this:
```js
SUPPORTED_METHODS.forEach(function (method) {
  var override = overrides[method];
  if (typeof override === 'function') {
    logger[method] = override.bind(overrides);
  }
});
```

Reply:
Why bind to the object of overrides rather than the function itself? Or actually, maybe more accurate would be:

```javascript
logger[method] = override.bind(logger);
```

Reply:
you can try these code  in chrome 51:
```js
var logger = {};
var override = console.log;
logger.log = override.bind(override);
logger.log('test')
```
and
```js
var logger = {}
var override = console.log
logger.log = override.bind(logger)
logger.log('test')
```
both throw error
![image](https://user-images.githubusercontent.com/5723181/57517660-ed25e000-7349-11e9-9634-66a0f960913e.png)

Reply:
but this is successful:
```js
var logger = {};
var override = console.log;
logger.log = override.bind(console);
logger.log('test');
```
![image](https://user-images.githubusercontent.com/5723181/57518361-6c67e380-734b-11e9-9183-7094aad3d222.png)

Reply:
You're getting that error because of [this issue with Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=179628). The correct thing to do is:

```javascript
var logger = {};
var override = console.log.bind(console);
logger.log = override;
logger.log('test');
```